### ISSUE TITLE
[youtube] "Please sign in" error when using `--live-from-start`

### ISSUE THREAD

**Issue by rrfqi** [yt-dlp/yt-dlp#11201]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Indonesia

### Provide a description that is worded well enough to be understood

sorry for making this new one since [the old one](https://github.com/yt-dlp/yt-dlp/issues/8450) already closed

is there any solution to bypass the sign in error? I've been always getting this error today while two days ago sometimes the sign in error came up but I still can manage to download after trying it couple times, but today it always getting `ERROR: [youtube] 8WCKaco64js: Please sign in`

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-v', '--external-downloader', 'ffmpeg', '--hls-use-mpegts', '--live-from-start', '--download-sections', '*1728286200-1728287200', '-o', 'test.%%(ext)s', 'https://www.youtube.com/watch?v=8WCKaco64js']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version bashonly/yt-dlp@live-from-sections build 2024.08.17.094430 [ded5847da] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-109138-g3bcec58535-20221119 (setts)
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.07.04, curl_cffi-0.5.10, mutagen-1.47.0, requests-2.32.3, sqlite3-3.35.5, urllib3-2.2.2, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1830 extractors
[youtube] Extracting URL: https://www.youtube.com/watch?v=8WCKaco64js
[youtube] 8WCKaco64js: Downloading webpage
[youtube] 8WCKaco64js: Downloading ios player API JSON
[youtube] 8WCKaco64js: Downloading web creator player API JSON
ERROR: [youtube] 8WCKaco64js: Please sign in
  File "yt_dlp\extractor\common.py", line 740, in extract
  File "yt_dlp\extractor\youtube.py", line 4398, in _real_extract
  File "yt_dlp\extractor\common.py", line 1263, in raise_no_formats
```

**Comment by grqz** [comment]
you need to update. if you dont need `live-from-sections`, you can just use https://github.com/yt-dlp/yt-dlp/releases/download/2024.10.07/yt-dlp.exe instead temporarily. you can also request `@`bashonly to merge `yt-dlp:master` into the branch(or merge master and build it yourself)

**Comment by rrfqi** [comment]
yeah, unfortunately the latest yt-dlp version still not supported for download-sections

> [debug] Command-line config: ['-v', '--external-downloader', 'ffmpeg', '--hls-use-mpegts', '--live-from-start', '--download-sections', '*1728286200-1728287200', '-o', 'test.%%(ext)s', 'https://www.youtube.com/watch?v=8WCKaco64js']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.10.07 from yt-dlp/yt-dlp [1a176d874] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-109138-g3bcec58535-20221119 (setts)
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.5.10, mutagen-1.47.0, requests-2.32.3, sqlite3-3.35.5, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1838 extractors
[youtube] Extracting URL: https://www.youtube.com/watch?v=8WCKaco64js
[youtube] 8WCKaco64js: Downloading webpage
[youtube] 8WCKaco64js: Downloading ios player API JSON
[youtube] 8WCKaco64js: Downloading mweb player API JSON
[youtube] 8WCKaco64js: Downloading MPD manifest
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 8WCKaco64js: Downloading 1 format(s): 137+140
[info] 8WCKaco64js: Downloading 1 time ranges: 1728286200.0-1728287200.0
ERROR: This format cannot be partially downloaded. Aborting
  File "yt_dlp\__main__.py", line 17, in <module>
  File "yt_dlp\__init__.py", line 1086, in main
  File "yt_dlp\__init__.py", line 1076, in _real_main
  File "yt_dlp\YoutubeDL.py", line 3609, in download
  File "yt_dlp\YoutubeDL.py", line 3582, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1615, in extract_info
  File "yt_dlp\YoutubeDL.py", line 1626, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1782, in __extract_info
  File "yt_dlp\YoutubeDL.py", line 1841, in process_ie_result
  File "yt_dlp\YoutubeDL.py", line 3015, in process_video_result
  File "yt_dlp\YoutubeDL.py", line 179, in wrapper
  File "yt_dlp\YoutubeDL.py", line 3384, in process_info
  File "yt_dlp\YoutubeDL.py", line 1092, in report_error
  File "yt_dlp\YoutubeDL.py", line 1020, in trouble

however this one `--extractor-args "youtube:player_client=ios,mweb"` does help with the problem, thank you

**Comment by bashonly** [comment]
Fixed by de2062753a188060d76f587e45becce61fe399f9 (changing default Youtube player clients to `ios,mweb`)

Previous to that commit, the default clients were `ios,web_creator`. The `web_creator` client is now requiring sign-in, and our requests get the "please sign in" error in response (this is a different error than the "...to confirm your not a bot" error). If you didn't pass `--live-from-start`, yt-dlp would silently ignore this error and continue with the formats provided by the `ios` client. However, the `ios` client doesn't provide the necessary DASH formats for `--live-from-start`, so yt-dlp throws the error here. Related: #11165, #11185

**Note**: I'm making an exception here and treating this as a valid issue since we didn't previously have one to track this `web_creator` "please sign in" error. But in the future, please **do not** open issues if you're not using the **latest nightly/master** version of yt-dlp. Custom builds/forks/PR branches are not supported in the yt-dlp/yt-dlp issue tracker.

P.S. I made a new pre-release build:`yt-dlp --update-to bashonly/yt-dlp@live-from-sections`

**Comment by rrfqi** [comment]
> Fixed by [de20627](https://github.com/yt-dlp/yt-dlp/commit/de2062753a188060d76f587e45becce61fe399f9) (changing default Youtube player clients to `ios,mweb`)
> 
> Previous to that commit, the default clients were `ios,web_creator`. The `web_creator` client is now requiring sign-in, and our requests get the "please sign in" error in response (this is a different error than the "...to confirm your not a bot" error). If you didn't pass `--live-from-start`, yt-dlp would silently ignore this error and continue with the formats provided by the `ios` client. However, the `ios` client doesn't provide the necessary DASH formats for `--live-from-start`, so yt-dlp throws the error here. Related: #11165, #11185
> 
> **Note**: I'm making an exception here and treating this as a valid issue since we didn't previously have one to track this `web_creator` "please sign in" error. But in the future, please **do not** open issues if you're not using the **latest nightly/master** version of yt-dlp. Custom builds/forks/PR branches are not supported in the yt-dlp/yt-dlp issue tracker.
> 
> P.S. I made a new pre-release build:`yt-dlp --update-to bashonly/yt-dlp@live-from-sections`

ah I see alright, thank you very much
